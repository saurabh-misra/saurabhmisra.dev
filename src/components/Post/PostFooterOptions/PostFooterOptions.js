import React from "react";
import { useSiteMetadata } from "../../../hooks";
import { getDiscussURL } from "../../../utils";
import styles from "./PostFooterOptions.module.scss";

const PostFooterOptions = ({ postSlug }) => {
  const { url } = useSiteMetadata();
  const twitterDiscussUrl = getDiscussURL(url, postSlug);

  return (
    <div className={styles["options"]}>
      <a href={twitterDiscussUrl} target="_blank">
        Discuss on Twitter
      </a>
    </div>
  );
};

export default PostFooterOptions;
