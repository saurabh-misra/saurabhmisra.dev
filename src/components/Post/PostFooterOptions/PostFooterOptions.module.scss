@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.options {
  border-top: 1px solid #e6e6e6;
  padding-top: 1.25rem;

  &__links {
    text-decoration: underline;
  }
}
