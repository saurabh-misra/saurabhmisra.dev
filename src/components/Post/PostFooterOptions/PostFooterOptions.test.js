// @flow
import React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery, StaticQuery } from "gatsby";
import PostFooterOptions from "./PostFooterOptions";
import siteMetadata from "../../../../jest/__fixtures__/site-metadata";
import type { RenderCallback } from "../../../types";

describe("PostFooterOptions", () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it("renders correctly", () => {
    const tree = renderer.create(<PostFooterOptions />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
