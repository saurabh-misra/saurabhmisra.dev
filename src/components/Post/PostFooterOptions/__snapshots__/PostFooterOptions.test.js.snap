// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFooterOptions renders correctly 1`] = `
<div
  className="options"
>
  <a
    href="https://mobile.twitter.com/search?q=http://localhostundefined"
    target="_blank"
  >
    Discuss on Twitter
  </a>
</div>
`;
